src/main/docs/guide/generator/usage.adoc

```diff
@@ -0,0 +1,50 @@
+The source generation from JSON schema can also be used during run-time with the `io.micronaut.jsonschema.generator` package. The following code snippet shows how `SourceGenerator` object can be used to generate beans of your desired language from JSON schema.
+
+[source,java]
+----
+// create a generator with your chosen language
+var javaGenerator = new SourceGenerator(VisitorContext.Language.JAVA);
+
+// Optional: can configure accepted URL references
+UrlLoader.setAllowedUrlPatterns(List.of("^https://.*/.*.schema.json$"));
+
+// Optional: set up input file name
+String schemaFileName = "example.schema.json";
+SourceGenerator.setInputFileName(schemaFileName);
+
+Path outputPath = Paths.get("output"); // Define the base output path
+String packageName = "com.example.temp"; // Example package name
+var config = new SourceGeneratorConfig(
+    null, "https://www.jsonschemastore.org/example/temp.schema.json",
+    null, null,
+    outputPath, packageName, null);
+
+javaGenerator.generate(config);
```

Comment:
replace with a snippet read from a test to ensure this doesn't break in the future